### PR TITLE
Sorting and checkout page basket editing

### PR DESCRIPTION
**Pull request by wortley**
- Added 3 modes of sorting to homepage
- Add ability to remove items from basket and edit their quantity on checkout page (+ add confirmations for removing items)
- Disable pay button if basket is empty
- Small grammar fixes (e.g. unit -> units for more than one unit of something)
- Fix bug with product card where if quantity set to 0 by typing it in, it was not set to 0 in basket

### CODE REVIEW

File: src/js/checkout.ts [hunk 1]
```diff
@@ -143,35 +145,87 @@ function updateTotalPrice() {
...
 
 function updateCheckoutList() {
   const checkoutList = document.querySelector('.checkoutList tbody') as HTMLTableElement
   const basketItemRows: Array<HTMLTableRowElement> = []
 
   if (basket.size === 0) {
-    (document.querySelector('#clearbasket') as HTMLButtonElement).disabled = true
+    (document.querySelector('#clearbasket') as HTMLButtonElement).disabled = true;
+    (document.querySelector('#paycreditcard') as HTMLButtonElement).disabled = true
+    document.querySelector('.checkoutList tbody')!.innerHTML = EMPTY_BASKET_HTML
```
Comment (Xanonymous-GitHub) [inline src/js/checkout.ts]:
```suggestion
    const checkoutListBody = document.querySelector('.checkoutList tbody')
    checkoutListBody?.replaceChildren();
    checkoutListBody?.insertAdjacentHTML('afterbegin', EMPTY_BASKET_HTML)
```

Suggest avoiding using `innerHTML` for security concern.

File: src/js/shop.ts [hunk 2]
```diff
@@ -163,3 +173,18 @@ async function asyncMakeProductCardElements() {
     return document.createElement('div')
   }))
 }
+
+async function sortAndUpdateProductCards(sortMethod: SORT_METHOD) {
```
Comment (Xanonymous-GitHub) [inline src/js/shop.ts:177]: curious about if it is necessarily to solve the case when both sorting method and search key are applied...

Comment (wortley) [reply src/js/shop.ts:177]: Added call to onSearchSubmitted

**Review by Xanonymous-GitHub — APPROVED** [review]
LGTM.

**Pull request merged**